test/NVWS/insert_aref.mlir

```diff
@@ -0,0 +1,88 @@
+// RUN: triton-opt %s -split-input-file -allow-unregistered-dialect --tritongpu-hoist-tmem-alloc -tritongpu-partition-scheduling --nvws-insert-aref | FileCheck %s
+
+#blocked = #ttg.blocked<{sizePerThread = [1, 128], threadsPerWarp = [32, 1], warpsPerCTA = [4, 1], order = [0, 1]}>
+#blocked1 = #ttg.blocked<{sizePerThread = [1, 1], threadsPerWarp = [1, 32], warpsPerCTA = [2, 2], order = [1, 0]}>
+#shared = #ttg.nvmma_shared<{swizzlingByteWidth = 128, transposed = false, elementBitWidth = 16}>
+#shared1 = #ttg.nvmma_shared<{swizzlingByteWidth = 128, transposed = true, elementBitWidth = 16}>
+#smem = #ttg.shared_memory
+#tmem = #ttng.tensor_memory_encoding<blockM = 128, blockN = 128, unpacked = true>
+
+module attributes {"ttg.num-warps" = 4 : i32, ttg.target = "cuda:100"} {
+  // FUNC-LABEL: @warp_specialize_tma_matmul
+  // CHECK: @warp_specialize_tma_matmul
+  tt.func @warp_specialize_tma_matmul(%arg0: i32, %arg1: i32, %arg2: i32, %arg3: !tt.tensordesc<tensor<128x64xf16, #shared>>, %arg4: !tt.tensordesc<tensor<128x64xf16, #shared>>) {
+    %true = arith.constant true
+    %c0_i32 = arith.constant 0 : i32
```

Comment:
what happens when the result of a descriptor load is used both in memory by a matmul and used as a register value?

Reply:
I believe we have a real case where this occurs. I think there is a unit test for it.

Reply:
Supported in the commit https://github.com/triton-lang/triton/pull/7581/commits/ae6cc0235c38404fc8c0c20afff276896676caaf